File: client/src/app/_dtos/process/deploymentPendingDto.ts

export interface IDeploymentPendingDto {
    depId: number;
    checked: boolean;
    cvRefId: number;
    referredOn: Date;
    selectedOn: Date;
    applicationNo: number;
    orderItemId: number;
    candidateName: string;
    customerName: string;
    cityOfWorking: string;
    orderNo: number;
    orderDate: Date;
    categoryName: string;
    currentStatus: string;

    ecnr: boolean;
    deploySequence: number;
    nextSequence: number;
    currentSeqDate: Date;
    nextStageDate: Date;

}

export class DeploymentPendingDto implements IDeploymentPendingDto {
    depId: number=0;
    checked: boolean=false;
    cvRefId: number=0;
    referredOn = new Date();
    selectedOn = new Date();
    orderItemId=0;
    applicationNo: number=0;
    candidateName: string='';
    ecnr=false;
    currentStatus = '';
    
    customerName: string='';
    cityOfWorking: string='';
    orderNo: number=0;
    orderDate: Date=new Date();
    categoryName: string='';
    
    selDecisionDate: Date=new Date();
    deploySequence: number=0;
    currentSeqDate = new Date('1900-01-01');
    nextSequence: number=0;
    nextStageDate: Date=new Date();

}

File: api/DTOs/Process/DeploymentPendingDto.cs
namespace api.DTOs.Process
{
    public class DeploymentPendingDto
    {
        public int DepId { get; set; }
        public int ApplicationNo {get; set;}
        public string CandidateName { get; set; }
        
        public DateTime TransactionDate { get; set; }
        public int CvRefId { get; set; }
        public int OrderItemId { get; set; }
        public bool Ecnr { get; set; }
        public string CustomerName { get; set; }
        public int CustomerId { get; set; }
        public string CityOfWorking { get; set; }
        public int OrderNo { get; set; }
        public DateTime OrderDate { get; set; }
        public string CategoryName { get; set; }
        public DateTime ReferredOn { get; set; }
        public DateTime SelectedOn {get; set;}
        public int DeploySequence { get; set; }
        public int NextSequence { get; set; }
        public DateTime CurrentSeqDate { get; set; }
        public DateTime NextStageDate { get; set; }
        public string CurrentStatus { get; set; }  
     }
}